# HG changeset patch
# User Vytautas Janƒçauskas <unaudio@gmail.com>
# Date 1378923930 -10800
#      Wed Sep 11 21:25:30 2013 +0300
# Node ID 23833e1bfa2e7705fd3665420c82febb8c17c6ba
# Parent  ea5c156aa804c8800788f2d1660bc87402998b59
check for PortAudio

* configure.ac: fix a typo in sndfile warning message, added checks for
PortAudio and relevant build variable exports

diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -935,17 +935,17 @@ LIBS="$save_LIBS"
 CPPFLAGS="$save_CPPFLAGS"
 
 ### Check for sndfile
 
 SNDFILE_CPPFLAGS=
 SNDFILE_LDFLAGS=
 SNDFILE_LIBS=
 
-warn_sndfile="$sndfile library fails tests.  The audioinfo, audioread and audiowrite functions for reading and writing audio files will not be fully functional."
+warn_sndfile="sndfile library fails tests.  The audioinfo, audioread and audiowrite functions for reading and writing audio files will not be fully functional."
 
 PKG_CHECK_EXISTS([sndfile], [
   SNDFILE_CPPFLAGS=`$PKG_CONFIG --cflags-only-I sndfile`
   SNDFILE_LDFLAGS=`$PKG_CONFIG --libs-only-L sndfile`
   SNDFILE_LIBS=`$PKG_CONFIG --libs-only-l sndfile`
   warn_sndfile=
 ])
 
@@ -957,16 +957,44 @@ else
   SNDFILE_LDFLAGS=
   SNDFILE_LIBS=
 fi
 
 AC_SUBST(SNDFILE_CPPFLAGS)
 AC_SUBST(SNDFILE_LDFLAGS)
 AC_SUBST(SNDFILE_LIBS)
 
+### Check for PortAudio
+
+PORTAUDIO_CPPFLAGS=
+PORTAUDIO_LDFLAGS=
+PORTAUDIO_LIBS=
+
+warn_portaudio="PortAudio library fails tests.  The audioplayer, audiorecorder classes and audiodevinfo function for audio playback and recording will not be fully functional."
+
+PKG_CHECK_EXISTS([portaudio-2.0 >= 19], [
+  PORTAUDIO_CPPFLAGS=`$PKG_CONFIG --cflags-only-I portaudio-2.0`
+  PORTAUDIO_LDFLAGS=`$PKG_CONFIG --libs-only-L portaudio-2.0`
+  PORTAUDIO_LIBS=`$PKG_CONFIG --libs-only-l portaudio-2.0`
+  warn_portaudio=
+])
+
+if test -z "$warn_portaudio"; then
+  AC_DEFINE(HAVE_PORTAUDIO, 1,
+    [Define to 1 if PortAudio is available.])
+else
+  PORTAUDIO_CPPFLAGS=
+  PORTAUDIO_LDFLAGS=
+  PORTAUDIO_LIBS=
+fi
+
+AC_SUBST(PORTAUDIO_CPPFLAGS)
+AC_SUBST(PORTAUDIO_LDFLAGS)
+AC_SUBST(PORTAUDIO_LIBS)
+
 ### Check for either of Graphics/ImageMagick++ libraries
 
 AC_ARG_WITH([magick],
   [AS_HELP_STRING([--with-magick=LIB],
     [select library to use for image I/O (options: GraphicsMagick(default) or ImageMagick)])],
   [magick="$withval"],
   [magick="GraphicsMagick"])
 
@@ -2876,16 +2904,19 @@ Octave is now configured for $canonical_
   LAPACK libraries:            $LAPACK_LIBS
   LLVM CPPFLAGS:               $LLVM_CPPFLAGS
   LLVM LDFLAGS:                $LLVM_LDFLAGS
   LLVM libraries:              $LLVM_LIBS
   Magick++ CPPFLAGS:           $MAGICK_CPPFLAGS
   Magick++ LDFLAGS:            $MAGICK_LDFLAGS
   Magick++ libraries:          $MAGICK_LIBS
   OPENGL libraries:            $OPENGL_LIBS
+  PortAudio CPPFLAGS:          $PORTAUDIO_CPPFLAGS
+  PortAudio LDFLAGS:           $PORTAUDIO_LDFLAGS
+  PortAudio libraries:         $PORTAUDIO_LIBS
   PTHREAD flags:               $PTHREAD_CFLAGS
   PTHREAD libraries:           $PTHREAD_LIBS
   QHULL CPPFLAGS:              $QHULL_CPPFLAGS
   QHULL LDFLAGS:               $QHULL_LDFLAGS
   QHULL libraries:             $QHULL_LIBS
   QRUPDATE CPPFLAGS:           $QRUPDATE_CPPFLAGS
   QRUPDATE LDFLAGS:            $QRUPDATE_LDFLAGS
   QRUPDATE libraries:          $QRUPDATE_LIBS
